source/dotnet/Library/AdaptiveCards/StringSizeWithUnitConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace AdaptiveCards
+{
+    public class StringSizeWithUnitConverter : JsonConverter, ILogWarnings
+    {
+        public List<AdaptiveWarning> Warnings { get; set; } = new List<AdaptiveWarning>();
+
+        readonly JsonSerializer defaultSerializer = new JsonSerializer();
+
+        public override bool CanConvert(Type objectType)
+        {
+            // Only use this converter for string types that match our format
+            return (objectType == typeof(string));
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType == JsonToken.String)
+            {
+                var dimension = defaultSerializer.Deserialize(reader) as string;
+                if(uint.TryParse(dimension.Substring(0, dimension.Length - 2), out uint dimensionInPix)) {
+                    return dimensionInPix;
```

Comment:
This does not validate that the unit is properly specified. It just ignores the last 2 chars.

The following renders as 50x50:
            "type": "Image",
            "url": "[http://adaptivecards.io/content/cats/1.png](http://adaptivecards.io/content/cats/1.png)",
            "width": "500t",
            "height": "500t"

Unit must be pt lowercase, otherwise ignore it and set 0.

Reply:
thank you; 

---
In reply to: [213858594](https://github.com/Microsoft/AdaptiveCards/pull/1840#discussion_r213858594) [](ancestors = 213858594)